File: Client/typescript/module/seasonTableView/SeasonTableViewModel.ts
module FifaLeagueClient.Module.SeasonTableView {


    export class SeasonTableViewModel {

    public Id: number;
    public Name:String;
    public RuleSet:FifaLeagueClient.Module.Rules.RulesSetModel;
    public LeagueTables: LeagueTableViewModel[];

    // build the model directly from the data returned by the service
    constructor(data){
        if(data != null){
            this.Id = data.Id;
            this.Name = data.Name;
            this.LeagueTables = data.LeagueTables;
            this.RuleSet = data.RuleSet;
        }
    }
}

    export class LeagueTableViewModel {
        public Id: number;
        public Name:String;
        public TeamPlayers: TeamPlayerTableLeagueViewModel[];

        // build the model directly from the data returned by the service
        constructor(data){
                if(data != null){
                        this.Id = data.Id;
                        this.Name = data.Name;
                        this.TeamPlayers = data.TeamPlayers;
                }
        }
    }

    export class TeamPlayerTableLeagueViewModel {

            public Id: number;
            public position: number;
            public nbPlayedMatches: number;
            public nbGoalsFor: number;
            public nbGoalsAgainst: number;
            public nbGoalsDiff: number;
            public nbWin: number;
            public nbDraw: number;
            public nbLost: number;
            public nbPoints: number;
            public player:Player.PlayerModel;
            public team: Team.TeamModel;
            public show:boolean;

            // build the model directly from the data returned by the service
            constructor(data){
                    if(data != null){
                            this.position = data.position;
                            this.nbPlayedMatches = data.nbPlayedMatches;
                            this.nbGoalsFor = data.nbGoalsFor;
                            this.nbGoalsAgainst = data.nbGoalsAgainst;
                            this.nbGoalsDiff = data.nbGoalsDiff;
                            this.nbWin = data.nbWin;
                            this.nbDraw = data.nbDraw;
                            this.nbLost = data.nbLost;
                            this.nbPoints = data.nbPoints;
                            this.player = data.player;
                            this.team = data.team;
                            this.show = false;
                    }
            }
    }


}

File: Server/FIFA.Server/Models/SeasonTable/SeasonTableViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace FIFA.Server.Models
{

    public class TeamPlayerTableLeagueViewModel
    {
        public int Id { get; set; }
        public int position { get; set; }
        public int matchId { get; set; }
        public int seasonId { get; set; }
        public string seasonName { get; set; }
        public int leagueId { get; set; }
        public string leagueName { get; set; }
        public int nbPlayedMatches { get; set; }
        public int nbGoalsFor { get; set; }
        public int nbGoalsAgainst { get; set; }
        public int nbGoalsDiff { get; set; }
        public int nbWin { get; set; }
        public int nbDraw { get; set; }
        public int nbLost { get; set; }
        public int nbPoints { get; set; }
        public Player player { get; set; }
        public Team team { get; set; }

    }

    // Class showing a league with all the players and the teams
    public class LeagueTableViewModel
    {
        public int seasonId { get; set; }
        public string seasonName { get; set; }
        public int Id { get; set; }
        public string Name { get; set; }
        public IEnumerable<TeamPlayerTableLeagueViewModel> TeamPlayers { get; set; }
    }


    // Class showing a season with all the leagues
    public class SeasonTableViewModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public IEnumerable<LeagueTableViewModel> LeagueTables { get; set; }
    }
}